staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_test.go

```diff
@@ -425,14 +430,20 @@ func withSpecNodeNameIndexerFuncs(options *setupOptions) {
 			return pod.Spec.NodeName
 		},
 	}
+	options.indexers = map[string]cache.IndexFunc{
+		"f:spec.nodeName": func(obj interface{}) ([]string, error) {
```

Comment:
I can't remember, do indexers have to deal with DeletedFinalStateUnknown objects?

Reply:
Not sure, first time I hear about DeletedFinalStateUnknown object. The reason for this change is fact that cacher tests never used indexers because they never configured them. Added testing in https://github.com/kubernetes/kubernetes/pull/126875

Reply:
verified we don't need to handle that ... when handling a delete event the old obj is pulled from the cache and the new obj is forced to nil:

https://github.com/kubernetes/kubernetes/blob/5ac315faf428567ec023aa7d8c2b89b1d9c9211a/staging/src/k8s.io/client-go/tools/cache/thread_safe_store.go#L248

so the index func is not invoked on the new object at all